traverse.go

```diff
@@ -17,22 +19,43 @@ type Networks struct {
 	err      error
 }
 
+var allIPv4 = &net.IPNet{IP: make(net.IP, 4), Mask: net.CIDRMask(0, 32)}
+var allIPv6 = &net.IPNet{IP: make(net.IP, 16), Mask: net.CIDRMask(0, 128)}
+
 // Networks returns an iterator that can be used to traverse all networks in
 // the database.
 //
 // Please note that a MaxMind DB may map IPv4 networks into several locations
 // in an IPv6 database. This iterator will iterate over all of these
 // locations separately.
 func (r *Reader) Networks() *Networks {
-	s := 4
+	var networks *Networks
 	if r.Metadata.IPVersion == 6 {
-		s = 16
+		networks = r.NetworksWithin(allIPv6)
+	} else {
+		networks = r.NetworksWithin(allIPv4)
 	}
+
+	return networks
+}
+
+func (r *Reader) NetworksWithin(network *net.IPNet) *Networks {
```

Comment:
This needs documentation. I think your earlier PR had documentation. Also, an example similar to this one for `Networks()` would be nice:

https://github.com/oschwald/maxminddb-golang/blob/7abc676b6f6c2bedd14889567679b0fd2a9fff06/example_test.go#L56

This would get included in the generated documentation.